lib/rack/multipart/parser.rb

```diff
@@ -265,8 +265,8 @@ def handle_mime_head
       end
 
       def handle_mime_body
-        if @sbuf.check_until(@body_regex) # check but do not advance the pointer yet
-          body = @sbuf[1]
+        if (body_with_boundary = @sbuf.check_until(@body_regex)) # check but do not advance the pointer yet
```

Comment:
In looking at this issue, have you seen why this line uses check instead of scan? So far I can't see why `on_mime_body` cares about the `pos`, and we update it on the next line. :confused:

Reply:
I don't understand the logic completely, but I guess it is the `else`-path of the check, that uses methods like `rest_size` and `peek`. I think the key question is, why this `rx_max_size` stuff is there.

I tried to make the changes as minimal as possible to solve the issue, but maybe we could refactor some of the other stuff, too.